### DISCUSSION TITLE
Query connectOrCreate

### DISCUSSION THREAD

**Discussion by joao99jobs** [prisma/prisma#18341] [Q&A] [answered]
Hey guys!

I'm trying to do a create query and I don't think I know how to structure it correctly. I want to create a post, and that post is required to link to a book, but not required to be linked to a collection.

<img width="612" alt="image" src="https://user-images.githubusercontent.com/47746137/225485692-d4e7bea9-c1ce-462d-a4a0-866f10d3b8a7.png">

<img width="375" alt="image" src="https://user-images.githubusercontent.com/47746137/225485719-773943e3-929f-4911-a89d-fd9215e58025.png">

But I am getting this error:
<img width="1421" alt="image" src="https://user-images.githubusercontent.com/47746137/225486230-8f2a8c1f-e19b-41ef-9455-b80f2f9ba6d0.png">

Remembering that I'm purposely passing collection as null because I didn't select any collection in the front-end.
If there's no way to do this the way I'd like, do you have a suggestion?

**Comment by nurul3101** [comment, marked as answer]
Hello @joao99jobs 👋 

You are getting this error because the value of `collection` variable is null which is not a valid value.

To avoid this, you can add a condition before invoking the create query, if the value of collection is null then you don't need to add `connectOrCreate` clause and only when collection is not null you add a `connectOrCreate` clause.

**Reply by joao99jobs** [reply]
Thank you very much @nurul3101 !

**Reply by nurul3101** [reply]
No worries! 🙂
